Update PHP version to 8.1 and other updates to `helpers.sh`

jazzsequence opened the pull request:
This PR updates the `helpers.sh` script's `update_php` function to default to a minimum of PHP 8.1 since Sage no longer supports 8.0.

The update_php function has therefore been updated to check for a minimum of 8.1 but _additionally_ updates have been made so that `update_php` can be called directly from the script outside of the sage install workflow, so we can test _that component_ separately. 

BATS tests have been created to test the `update_php` function for all versions from 7.4 - 8.3.

This PR also makes the following minor(ish) adjustments to the sage install script workflow:
- normalizes the generated theme names before the sage install script is called (fixes an issue where tests were failing because themes were created that included slashes in the directories that were created)
- silences the multisite check (sends the output of the check to `/dev/null` so a WP-CLI warning is not thrown if the `MULTISITE` constant does not exist)
- adds a check to see if the Sage theme exists before attempting to activate it

jazzsequence committed:
set the default php version to 8.1

jazzsequence committed:
update the update_php function to update to 8.1 at minimum

jazzsequence committed:
allow the update_php function to be run independently
so we can test it

jazzsequence committed:
add tests for the updated update_php function

jazzsequence committed:
update the test script to add bats tests

jazzsequence committed:
update the echo to be more accurate

jazzsequence committed:
add more verbose inline docs

jazzsequence committed:
bail if we can't copy the file

jazzsequence committed:
ensure the job runner has permission to write to the filesystem

jazzsequence committed:
update the sed command to work cross-platform
maybe....

jazzsequence committed:
don't bail out of the script entirely in the update php script if in CI

jazzsequence committed:
bail early (and don't try to activate) if the theme wasn't found

jazzsequence committed:
remove line breaks or spaces from the theme name

jazzsequence committed:
echo the themelist so we can see what's being grepped

jazzsequence committed:
change the grep to be more inclusive

jazzsequence committed:
use grep -q instead of -w

jazzsequence committed:
normalize genrenated theme names

pwtyler commented on private/scripts/helpers.sh:
```diff
@@ -476,17 +476,24 @@ function clean_up() {
   if ! terminus wp -- "$sitename"."$siteenv" theme activate "$sagename"; then
+    terminus wp -- "$sitename"."$siteenv" theme list
     echo "${red}Theme activation failed. Exiting here.${normal}"
     echo "Check the theme list above. If the theme you created is not listed, it's possible that the deploy has not completed. You can try again in a few minutes using the following command:"
     echo "terminus wp -- $sitename.dev theme activate $sagename"
```
Is this still accurate if we're exiting above in the case its not found

jazzsequence replied:
Probably not. 😁

jazzsequence committed:
move the messaging about the theme not being in the list up to that check
we can also remove the conditional around activation because if it's in the list, we should be able to activate it
